public/upload.js

```diff
@@ -0,0 +1,147 @@
+function onChange(event) {
```

Comment:
there's another file `frontend/src/upload.js` which I believe this is based off of. they should probably be combined.